url5583f3dbb2_XOR19_XScript_tgz-pJ8-xscript_interactive_MainJ8/src/xscript/compiler/treemaker/XTreeMaker.java

```diff
@@ -0,0 +1,1486 @@
+	}
+
+	private void skip(){
+		if(skip){
+			skip = false;
+			XToken token = tokenizer.getToken();
+			while(token.kind==XTokenKind.KEYWORD){
+				switch((XKeyword)token.data){
+				case AND:
+				case CASE:
+				case CATCH:
+				case COLON:
+				case COMMA:
+				case DEFAULT:
+				case DIV:
+				case DOT:
+				case ELSE:
+				case EQUAL:
+				case FINALLY:
+				case GREATER:
+				case IDIV:
+				case INSTANCEOF:
+				case ISSUBCLASS:
+				case MOD:
+				case MUL:
+				case OR:
+				case QUESTIONMARK:
+				case RBRAKET:
+				case RINDEX:
+				case RSCOPE:
+				case SAND:
+				case SBAND:
+				case SBOR:
+				case SMALLER:
+				case SMOD:
+				case SOR:
+				case SPOW:
+				case SXOR:
+				case XOR:
+				case AS:
+					break;
+				default:
+					return;
+				}
+				token = tokenizer.next();
+			}
+		}
+	}
+	
+	public XTreeModule makeModule() {
+		tokenizer.next();
+		startPosition();
+		List<XTreeStatement> list = new ArrayList<XTreeStatement>();
+		while(!tokenEof()){
+			list.add(makeStatement());
+			skip();
+		}
+		XTreePosition position = endPosition();
+		XTreeScope scope = new XTreeScope(position.clone(), list);
+		return new XTreeModule(position, scope);
+	}
+	
+	public XTreeModule makeModuleInteractive() {
+		tokenizer.next();
+		startPosition();
+		List<XTreeStatement> list = new ArrayList<XTreeStatement>();
+		while(!tokenEof()){
+			list.add(makeStatementInteractive());
+			skip();
+		}
+		XTreePosition position = endPosition();
+		XTreeScope scope = new XTreeScope(position.clone(), list);
+		return new XTreeModule(position, scope);
+	}
+	
+	public XTreeLowExpr makeExprOrDecl(){
+		XToken token = tokenizer.getToken();
+		if(token.kind==XTokenKind.KEYWORD){
+			startPosition();
+			switch((XKeyword)token.data){
+			case CLASS:
+				return makeClassPosStarted(true);
+			case FUNC:
+				return makeFuncPosStarted(true);
+			case GLOBAL:{
+				tokenizer.next();
+				if(tokenOk(XTokenKind.KEYWORD, XKeyword.CLASS)){
+					return makeClassPosStarted(true, XVarAccess.GLOBAL);
+				}else if(tokenOk(XTokenKind.KEYWORD, XKeyword.FUNC)){
+					return makeFuncPosStarted(true, XVarAccess.GLOBAL);
+				}
+				List<XTreeVarDeclEntry> entries = makeDeclEntries();
+				return new XTreeVarDecl(endPosition(), entries, XVarAccess.GLOBAL);
+			}case LOCAL:{
+				tokenizer.next();
+				if(tokenOk(XTokenKind.KEYWORD, XKeyword.CLASS)){
+					return makeClassPosStarted(true, XVarAccess.LOCAL);
+				}else if(tokenOk(XTokenKind.KEYWORD, XKeyword.FUNC)){
+					return makeFuncPosStarted(true, XVarAccess.LOCAL);
+				}
+				List<XTreeVarDeclEntry> entries = makeDeclEntries();
+				return new XTreeVarDecl(endPosition(), entries, XVarAccess.LOCAL);
+			}case NONLOCAL:{
+				tokenizer.next();
+				List<XTreeVarDeclEntry> entries = makeDeclEntries();
+				return new XTreeVarDecl(endPosition(), entries, XVarAccess.NONLOCAL);
+			}
+			default:
+				break;
+			}
+			endPosition();
+		}
+		return makeExpr();
+	}
+	
+	public XTreeStatement makeStatement() {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 157 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 99% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 355-361 into a separate method.